lxd/storage/drivers/driver_dir_patches.go

```diff
@@ -0,0 +1,60 @@
+package drivers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/lxc/lxd/lxd/db"
+)
+
+func (d *dir) patchBucketNames() error {
+	var buckets map[string]*db.StorageBucket
+
+	err := d.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
```

Comment:
Please use `d.state.ShutdownCtx`.

Reply:
In this case it does make sense because this transaction is only loading rows.
I would be cautious generally cancelling a transaction halfway through in a patch as we don't want to get into a state where (if we were doing multiple transactions) that a patch is partially applied.

But here its fine.